src/logfile_values.py

```diff
@@ -7,42 +7,298 @@
+
+    if segment is not None:
+        yield return_metrics_values(segment, metrics, time_format, columns_num)
+
+
+def convert_to_timestamp(time_str, time_format):
+    try:
+        timestamp = int(
+            time.mktime(time.strptime(time_str.split('+', 1)[0], time_format)))
+    except ValueError:
+        try:
+            timestamp = int(time_str)
+        except ValueError:
+            timestamp = int(
+                time.mktime(
+                    time.strptime('-'.join(time_str.split('-')[:-1]),
+                                  time_format)))
+    return int(timestamp)
+
+
 def main():
     args = parse_args()
-    template = args.prefix + '.{} {} ' + str(int(time()))
-    with open(args.file, 'r') as f:
-        for line in f:
-            pass
-        last_line = line.split()
+    if args.debug:
+        logging.getLogger().setLevel(logging.DEBUG)
+        logging.getLogger().addHandler(logging.StreamHandler())
+
+    if args.time_zone:
+        os.environ['TZ'] = args.time_zone
+        time.tzset()
+
+    file_was_readed = True
+
+    # Read from the end of file until the timestamp is satisfying conditions
+    data = read_logfile_reverse(
+        args.file, args.columns_num, args.time_format, metrics=args.metric)
+    for log_value in data:
+        # Stop reading if next line has bigger timestamp as a required
+        if not log_value:
+            file_was_readed = log_value
+            break
+        else:
+            for index, value in enumerate(log_value):
+                if value != -1:
+                    args.metric[index].values.append(value)
+
+    # If the main file was read completely and there is arch flag then
+    # check archive files for the presence of a timestamp satisfying condition
+    if args.arch and file_was_readed:
+        dir_path = os.path.dirname(os.path.realpath(args.file))
+        archive_pattern = re.compile(r'{}\.\d+?\.gz'.format(args.file))
+        for root, dirs, files in os.walk(dir_path):
+            for f in files:
+                if archive_pattern.search(f):
+                    archive_file = os.path.join(root, f)
+                    logging.getLogger().debug(
+                        'Parsing archive file: {}'.format(f))
+                    with gzip.open(archive_file, 'rt', encoding='utf-8') as fh:
+                        for line in fh:
+                            fields = line.split()
+                            if len(fields) != 5:
+                                continue
+                            fields[0] = convert_to_timestamp(
+                                fields[0], args.time_format)
+                            for metric in args.metric:
+                                if fields[0] > (
+                                        metric.now - metric.get_timeshift()):
+                                    metric.values.append(
+                                        metric.estimate_columns_value(fields))
```

Comment:
This is still pretty complicated.  Any luck splitting it into a few more functions?